site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md

```diff
@@ -0,0 +1,469 @@
+  apiSuggestions.unshift(input);
+  apiSuggestions.splice(NUMBER_OF_PREVIOUS_SEARCHES);
+  await chrome.storage.local.set({ apiSuggestions });
+}
+```
+
+{% Aside 'important' %}
+
+Extension service workers have access to both web APIs and Chrome APIs, with a few exceptions.
+For a deep dive, see [Service Workers...](tbd) 
+
+{% endAside %}
+
+### Step 6: Set up a recurring event {: #step-6 }
+
+The `setTimeout()` or `setInterval()` methods are commonly used to perform delayed or periodic
+tasks. However, these APIs can fail because the scheduler will cancel the timers when the service
+worker is terminated. Instead, extensions can use the [`chrome.alarms`][api-alarms] API. 
+
+To use the Alarms API, request the `"alarms"` permission in the manifest. The extension also needs to request [host permission][doc-host-perm] to retrieve the extension tips from the glitch site:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json/2/3
+{
+  ...
+  "permissions": ["storage", "alarms"],
+  "permissions": ["storage"],
+  "host_permissions": ["https://extension-tips.glitch.me/*"],
+}
+```
+
+The following code sets up an alarm once a day to fetch the daily tip and save it to
+[`chrome.storage.local()`][api-storage]:
+
+{% Label %}sw-tips.js:{% endLabel %}
+
+```js
+// Fetch tip & save in storage
+const updateTip = async () => {
+  const response = await fetch('https://extension-tips.glitch.me/tips.json');
+  const tips = await response.json();
+  const randomIndex = Math.floor(Math.random() * tips.length);
+  await chrome.storage.local.set({ tip: tips[randomIndex] });
+};
+
+// Create a daily alarm and retrieves the first tip when extension is installed.
+chrome.runtime.onInstalled.addListener(({ reason }) => {
+  if (reason === 'install') {
+    chrome.alarms.create({ delayInMinutes: 1, periodInMinutes: 1440 });
+    updateTip();
+  }
+});
+
+// Update tip once a the day
+chrome.alarms.onAlarm.addListener(updateTip);
+
+```
+
+{% Aside %}
+
+All [Chrome API][doc-apis] event listeners and methods restart the service worker 30 second termination timer. See [Extension service worker lifecycle](tdb) to learn more.
+
+{% endAside %}
+
+### Step 7: Communicate with other contexts {: #step-7 }
+
+[Content scripts][doc-content] communicate with the rest of the extension through [message passing][doc-messages]. In this example, the content script will request the tip of the day from the service worker. 
+
+First, declare the content script in the manifest and add the match pattern corresponding to the [Chrome API][doc-apis] reference documentation.
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "content_scripts": [
+    {
+      "matches": ["https://developer.chrome.com/docs/extensions/reference/*"],
+      "js": ["content.js"]
+    }
+  ]
+}
+
+```
+
+Create a new content file. The following code generates a button that will open the tip popover. It also sends a message to the service worker requesting the extension tip.
```

Comment:
Maybe mention that content scripts can't access storage.local.

Reply:
Technically, they DO have access to the storage API, but for security reasons, we are encouraged to use messages instead.